snuba/clickhouse/query.py

```diff
@@ -12,12 +12,22 @@ def __init__(self,
         # body anymore.
         request: Request,
         prewhere_conditions: Sequence[str],
-        final: bool,
+        query_hints: Mapping[str, Any],
```

Comment:
Since the query hints abstraction does not come from json load, I don't think there is a reason to keep it a completely untyped mapping. You can have a class like Query, which makes it a lot safer to use.